torch/nn/functional.py

```diff
@@ -2869,7 +2869,8 @@ def kl_div(
         else:
             reduction_enum = _Reduction.get_enum(reduction)
 
-    reduced = torch.kl_div(input, target, reduction_enum, log_target=log_target)
+    expanded_input, expanded_target = torch.broadcast_tensors(input, target)
```

Comment:
I see that this matches what was done for e.g. `nn.MSELoss`. With manual broadcasting here, do we still need the kernel changes?

Reply:
No, not really. But I like to keep it consistent, since forward is using `input.numel`, not `target.numel`.